modules/message-lane/src/inbound_lane.rs

```diff
@@ -61,14 +62,28 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 		}
 
 		data.latest_confirmed_nonce = outbound_lane_data.latest_received_nonce;
+		// Firstly, remove all of the records where higher nonce <= new confirmed nonce
 		while data
 			.relayers
 			.front()
-			.map(|(nonce, _)| *nonce <= data.latest_confirmed_nonce)
+			.map(|(_, nonce_high, _)| *nonce_high <= data.latest_confirmed_nonce)
 			.unwrap_or(false)
 		{
 			data.relayers.pop_front();
 		}
+		// Secondly, update the next record with lower nonce equal to new confirmed nonce if needed.
+		// Note: There will be max. 1 record to update as we don't allow messages from relayers to overlap.
+		if data
+			.relayers
+			.front()
+			.map(|(nonce_low, _, _)| *nonce_low <= data.latest_confirmed_nonce)
+			.unwrap_or(false)
+		{
+			data.relayers
+				.front_mut()
+				.expect("already ensured front relayer exists in preceding check")
+				.0 = data.latest_confirmed_nonce + 1;
+		}
```

Comment:
```suggestion
		match data.relayers.front_mut() {
		   Some(ref mut (nonce_low, _, _)) if *nonce_low < data.latest_confirmed_nonce => {
		     *nonce_low = data.latest_confirmed_nonce + 1;
		   }
		   _ => {},
		}
```

I find this code much more readable.